### ISSUE TITLE
Usage of both `-ms-high-contrast` and `forced-colors`

### ISSUE THREAD

**Issue by lgriffee** [Shopify/polaris-react-archive#4969]
## Issue summary
Some of our components are using the [`-ms-high-contrast`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/-ms-high-contrast) media feature and others are using [`forced-colors`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors). We should probably be consistent about which we are using. 

`-ms-high-contrast` will be retired in favor of `forced-colors` and we should eventually switch to using `forced-colors`. However, there isn't widespread support for `forced-colors` right now (Safari, Internet Explorer, WebView Android, Chrome Android, and Safari on iOS do not currently support it). 

I'm not sure why some components are using `forced-colors` alone with such poor browser compatibility (without `-ms-high-contrast` as a backup)  and question whether all components should be using either `-ms-high-contrast` alone or both media features until there is more support for `forced-colors`.

**Comment by lgriffee** [comment]
We should address this! Keep it open 👍🏻

**Comment by lgriffee** [comment]
We should address this! Keep it open 👍🏻

**Comment by lgriffee** [comment]
We should address this! Keep it open 👍🏻
